src/rpcmining.cpp

```diff
@@ -606,13 +636,27 @@ Value getblocktemplate(const Array& params, bool fHelp)
     result.push_back(Pair("version", pblock->nVersion));
     result.push_back(Pair("previousblockhash", pblock->hashPrevBlock.GetHex()));
     result.push_back(Pair("transactions", transactions));
-    result.push_back(Pair("coinbaseaux", aux));
-    result.push_back(Pair("coinbasevalue", (int64_t)pblock->vtx[0].vout[0].nValue));
+    if (coinbasetxn) {
+        result.push_back(Pair("coinbasetxn", txCoinbase));
```

Comment:
It seems to be assumed that ``txCoinbase`` will have been initialized by line 615 when ``coinbasetxn`` is true. I think it would be more robust to initialize ``txCoinbase`` to NULL where it is declared, and then assert here that it is nonnull, to avoid undefined behaviour if there is some bug that causes the coinbase tx to be missing.